src/map/utils/charutils.cpp

```diff
@@ -3445,7 +3452,15 @@ namespace charutils
 
                 if ((CurSkill / 10) < (CurSkill + SkillAmount) / 10) // if gone up a level
                 {
-                    PChar->WorkingSkills.skill[SkillID] += 1;
+                    // Light/Dark Arts artificially boost certain skills, logic is too complex to duplicate, so just recalculate it all (similar to using the JA)
+                    if (ArtsBonusActive(PChar, SkillID))
```

Comment:
I dont know _anything_ about SCH, but what this says to me is that we're going to call `BuildingCharSkillsTable` whenever  `TrySkillUP` is called - presumably on every action or melee swing etc. if `ArtsBonusActive` is on.

There's no DB calls in there, it's just a bunch of math, that seems OK, but logically: when `ArtsBonusActive` falls off, shouldn't we be rebuilding the table _one more time_ to put it back at its original non-inflated state?

Reply:
there's actually more work done when light/dark arts is gained/lost (in `onEffectGain`/`onEffectLose`)

and this is only going to do the extra math if the skillup in question is relevant to the currently-active arts buff. Perhaps that function could be renamed for clarity, but it returns true if SkiiID is one of the skills boosted by a current arts effect